libraries/SITL/SIM_Scrimmage.cpp

```diff
@@ -0,0 +1,212 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  simulator connector for Scrimmage simulator
+*/
+
+#include "SIM_Scrimmage.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <algorithm>
+
+#include <AP_HAL/AP_HAL.h>
+
+extern const AP_HAL::HAL& hal;
+
+namespace SITL {
+
+Scrimmage::Scrimmage(const char *home_str, const char *_frame_str) :
+    Aircraft(home_str, _frame_str),
+    prev_timestamp_us(0),
+    recv_sock(true),
+    send_sock(true),
+    frame_str(_frame_str)
+{
+    // Set defaults for scrimmage-copter
+    if (strcmp(frame_str, "scrimmage-copter")== 0) {
+        mission_name_ = "arducopter.xml";
+        motion_model_ = "Multirotor";
+        visual_model_ = "iris";
+    }
+}
+
+void Scrimmage::set_config(const char *config)
+{
+    fdm_port = 5504 + instance*10;
+    printf("ArduPilot sending to scrimmage on port: %d\n", fdm_port);
+    printf("ArduPilot listening to scrimmage on port: %d\n", fdm_port+1);
+
+    // TODO: un-hard code IP address and allow port to be changed.
+    recv_sock.bind("127.0.0.1", fdm_port+1);
```

Comment:
it should use the set_interface_ports() function to address this and port mapping

Reply:
Missed this function previously, now fixed